### PR TITLE
PMM-9775: respect UTC in sparkline

### PR DESCRIPTION
**Pull request by IuriiMr**
PMM-9775

Feature Build:
 - https://github.com/Percona-Lab/pmm-submodules/pull/3201

### CODE REVIEW

File: pmm-app/package.json [hunk 1]
```diff
@@ -52,7 +52,7 @@
     "highlight.js": "^11.4.0",
     "history": "^4.10.1",
     "lodash": "^4.17.20",
-    "moment": "^2.26.0",
+    "moment": "^2.29.4",
```
Comment (ritbl) [inline pmm-app/package.json:55]: let's use the latest moment to avoid potential TZ issues

File: pmm-app/src/pmm-qan/panel/provider/provider.tsx [hunk 2]
```diff
@@ -8,11 +8,24 @@ const initialState = {} as QueryAnalyticsContext;
 
 export const QueryAnalyticsProvider = React.createContext<QueryAnalyticsContext>(initialState);
 
-export const UrlParametersProvider = ({ timeRange, children }) => {
+export const UrlParametersProvider = (props) => {
```
Comment (ritbl) [inline pmm-app/src/pmm-qan/panel/provider/provider.tsx:11]: Reformatted with prettier

File: pmm-app/src/pmm-qan/panel/provider/provider.tsx [hunk 3]
```diff
@@ -22,9 +35,18 @@ export const UrlParametersProvider = ({ timeRange, children }) => {
         to: value[1],
       },
     }),
-    setActiveTab: (value) => (state) => ({ ...state, openDetailsTab: value }),
-    highlightSparkline: (value) => (state) => ({ ...state, highlightedCoords: value }),
-    setLoadingDetails: (value) => (state) => ({ ...state, loadingDetails: value }),
+    setActiveTab: (value) => (state) => ({
```
Comment (ritbl) [inline pmm-app/src/pmm-qan/panel/provider/provider.tsx:38]: reformat

File: pmm-app/src/pmm-qan/panel/provider/provider.tsx [hunk 4]
```diff
@@ -135,13 +157,15 @@ export const UrlParametersProvider = ({ timeRange, children }) => {
   };
 
   const query = new URLSearchParams(window.location.search);
-  const searchRef = useRef<string| null>(null);
+  const searchRef = useRef<string | null>(null);
```
Comment (ritbl) [inline pmm-app/src/pmm-qan/panel/provider/provider.tsx:160]: reformat

File: pmm-app/src/pmm-qan/panel/provider/provider.tsx [hunk 5]
```diff
@@ -8,11 +8,24 @@ const initialState = {} as QueryAnalyticsContext;
 
 export const QueryAnalyticsProvider = React.createContext<QueryAnalyticsContext>(initialState);
 
-export const UrlParametersProvider = ({ timeRange, children }) => {
+export const UrlParametersProvider = (props) => {
+  let {
+    timeRange,
+    children,
+    timeZone,
```
Comment (ritbl) [inline pmm-app/src/pmm-qan/panel/provider/provider.tsx:15]: adding timeZone

File: pmm-app/src/pmm-qan/panel/provider/provider.tsx [hunk 6]
```diff
@@ -135,13 +157,15 @@ export const UrlParametersProvider = ({ timeRange, children }) => {
   };
 
   const query = new URLSearchParams(window.location.search);
-  const searchRef = useRef<string| null>(null);
+  const searchRef = useRef<string | null>(null);
   const [fromTimeMomentValue, setFromTimeMomentValue] = useState(ParseQueryParamDate.transform(
-    query.get('from') || 'now-12h', 'from',
-  ).utc().format('YYYY-MM-DDTHH:mm:ssZ'));
```
Comment (ritbl) [inline pmm-app/src/pmm-qan/panel/provider/provider.tsx:141]: do not convert to UTC

File: pmm-app/src/shared/components/helpers/time-parameters-parser.ts [hunk 7]
```diff
@@ -55,8 +55,9 @@ const getComplexTimeValue = (date, nowFunc, edge) => {
 };
 
 export class ParseQueryParamDate {
-  static transform(date: string, edge?: TimeEdge) {
-    const nowFunc = getTimezone() === 'utc' ? moment.utc : moment;
+  static transform(date: string, edge?: TimeEdge, timeZone?: string) {
```
Comment (ritbl) [inline pmm-app/src/shared/components/helpers/time-parameters-parser.ts:58]: take TZ from props grafana sets, not from cookies

File: pmm-app/src/shared/components/Elements/Charts/Sparkline/Sparkline.tsx [hunk 8]
```diff
@@ -105,7 +114,10 @@ export const Sparkline = ({
       const value = isMetricExists(appLoadPolygonChart[columnNumber][ykey])
         ? 0
         : appLoadPolygonChart[columnNumber][ykey];
-      const dateToShow = moment(appLoadPolygonChart[columnNumber][xkey]).format('YYYY-MM-DD HH:mm:ss');
+
+      const tzFn = timeZone === 'utc' ? moment.utc : moment;
+      const dateToShow = tzFn(appLoadPolygonChart[columnNumber][xkey])
+        .format('YYYY-MM-DD HH:mm:ss Z');
```
Comment (ritbl) [inline pmm-app/src/shared/components/Elements/Charts/Sparkline/Sparkline.tsx:121]: added TZ in output so that is clear what TZ is used

**Review by matejkubinec — APPROVED** [review]

**Review by dliakhov — APPROVED** [review]

**Review by ritbl — APPROVED** [review]

**Pull request merged**